epochStart/metachain/stakingDataProvider.go

```diff
@@ -120,6 +120,26 @@ func (sdp *stakingDataProvider) GetNodeStakedTopUp(blsKey []byte) (*big.Int, err
 	return ownerInfo.topUpPerNode, nil
 }
 
+// GetNumStakedNodes returns the total number of owner's staked nodes
+func (sdp *stakingDataProvider) GetNumStakedNodes(owner []byte) (int64, error) {
+	ownerInfo, ok := sdp.cache[string(owner)]
+	if !ok {
+		return 0, epochStart.ErrOwnerDoesntHaveNodesInEpoch
+	}
+
+	return ownerInfo.numStakedNodes, nil
+}
+
+// GetTotalTopUp returns owner's total top up
+func (sdp *stakingDataProvider) GetTotalTopUp(owner []byte) (*big.Int, error) {
+	ownerInfo, ok := sdp.cache[string(owner)]
```

Comment:
we have some mutexes for other gets. add here as well

Reply:
+1 on all gets from the cache

Reply:
This is now obsolete, since this func is removed in future refactor PR